src/Cake.Core.Tests/Unit/Scripting/Processors/LoadDirectiveProcessorTests.cs

```diff
@@ -69,6 +69,25 @@ public void Should_Process_Multiple_Script_References_Found_In_Source(string sou
             Assert.Equal("/Working/other.cake", result.Script.Includes[1].Path.FullPath);
         }
 
+        [Theory]
+        [InlineData("#l \"%CAKE_TEST_SCRIPT_PATH%/utils.cake\"")]
+        public void Should_Process_Environment_Variable_Found_In_Source(string source)
+        {
+            // Given
+            System.Environment.SetEnvironmentVariable("CAKE_TEST_SCRIPT_PATH", "/Working");
```

Comment:
This fails on posix systems. Haven't checked why, this should only affect process, but could be something with how mono handles processes.

Reply:
To be honest I'm not a fan of setting environment variables like this in a test. It works but is more or less the same thing as a shared mutable state.

Reply:
Agree doesn't "feel" ok. Don't know what would be better, picking existing known to exist? That doesn't feel 100% either.